### PR TITLE
Fixes [added] feature table regions 

### LINKED ISSUE #29: Implement Table.regions()
The Java client implements Table.regions() using the SampleRowKeys RPC to get the start/stop keys.  The happybase client should do so as well.

### PR DESCRIPTION
**Pull request by pchauhan-qlogic**
Fix #29

### CODE REVIEW

File: src/google/cloud/happybase/table.py [hunk 1]
```diff
@@ -143,19 +143,23 @@ def families(self):
...
-        raise NotImplementedError(
-            "The Cloud Bigtable API does not have a "
-            "concept of splitting a table into regions."
-        )
+        regions = []
+        start_key = b""
+        for sample_row_key in self._low_level_table.sample_row_keys():
+            end_key = sample_row_key.row_key
+            if start_key is not end_key:
+                regions.append(RegionLocation(start_key, end_key))
```
Comment (sduskis) [inline src/google/cloud/happybase/table.py]: Please use a `dict` here instead of `RegionLocation`. There's no need for `RegionLocation` at all, so please remove it as well.

File: system_tests/test_happybase.py [hunk 2]
```diff
@@ -877,3 +879,20 @@ def test_counter_dec(self):
...
+
+
+class TestTable_region(BaseTableTest):
+
+    def test_region(self):
+        ALT_TABLE_NAME = "table_with_split_key"
+        connection = Config.CONNECTION
+
+        connection.create_table(ALT_TABLE_NAME, {COL_FAM1: {}})
+        table = connection.table(ALT_TABLE_NAME)
```
Comment (sduskis) [inline system_tests/test_happybase.py]: Please create a `google-cloud-python` `Table` with `initial_split_keys`, so that you can get meaningful `regions` from the HappyBase `table.regions()` call.

File: src/google/cloud/happybase/table.py (hunk 1 above)
Comment (pchauhan-qlogic) [reply src/google/cloud/happybase/table.py]: removed

File: system_tests/test_happybase.py (hunk 2 above)
Comment (pchauhan-qlogic) [reply system_tests/test_happybase.py]: added system test with passed split key in table

File: system_tests/test_happybase.py [hunk 3]
```diff
@@ -877,3 +880,26 @@ def test_counter_dec(self):
         self.assertEqual(
             table.row(ROW_KEY1, columns=[COL1]), {COL1: _PACK_I64(-dec_value)}
         )
+
+
+class TestTable_region(BaseTableTest):
+    def test_region(self):
+        from google.cloud.bigtable.table import Table as _LowLevelTable
+
+        ALT_TABLE_NAME = "table_with_split_key"
```
Comment (sduskis) [inline system_tests/test_happybase.py]: I think that it would be safer to add a unique id at the end of the table name `+ unique_resource_id("-")`.  That will prevent collisions in the future.

File: unit_tests/test_table.py [hunk 4]
```diff
@@ -1441,6 +1475,11 @@ def read_rows(self, *args, **kwargs):
             curr_row_data = rows_dict.pop(row_key)
             yield curr_row_data
 
+    def sample_row_keys(self):
+        for row_key in self.initial_split_keys:
+            self.sample_row_keys_result.append(_MockSampleRowKey(row_key))
+        return self.sample_row_keys_result
```
Comment (tseaver) [inline unit_tests/test_table.py]: It would be clearer just to have the testcase initialize `sample_row_keys_result` and just return it (rather than computing it), which would also let you test for the case that the back-end returns an empty sequence (the trailing `b""` isn't guaranteed in the proto).

File: src/google/cloud/happybase/table.py [hunk 5]
```diff
@@ -143,19 +140,22 @@ def families(self):
...
-
-            Cloud Bigtable does not give information about how a table is laid
-            out in memory, so this method does not work. It is
-            provided simply for compatibility.
-
-        :raises: :class:`NotImplementedError <exceptions.NotImplementedError>`
-                 always
+        :rtype: list
+        :returns: A list of regions with start_key and end_key.
+                  region is dictionay with attribute start_key and end_key.
```
[the remark this answers is not in the document]
Comment (pchauhan-qlogic) [reply src/google/cloud/happybase/table.py]: done

File: src/google/cloud/happybase/table.py [hunk 6]
```diff
@@ -143,19 +140,22 @@ def families(self):
...
         """
-        raise NotImplementedError(
-            "The Cloud Bigtable API does not have a "
-            "concept of splitting a table into regions."
-        )
+        regions = []
+        start_key = b""
+        for sample_row_key in self._low_level_table.sample_row_keys():
+            end_key = sample_row_key.row_key
+            if start_key is not end_key:
```
[the remark this answers is not in the document]
Comment (pchauhan-qlogic) [reply src/google/cloud/happybase/table.py]: tried with another way https://github.com/googleapis/google-cloud-python-happybase/pull/70/commits/8220b6bf6454c43e5822743a89d10d0980505248

File: unit_tests/test_table.py [hunk 7]
```diff
@@ -1420,6 +1451,9 @@ def __init__(self, *args, **kwargs):
         self.read_row_result = None
         self.read_rows_calls = []
         self.read_rows_result = None
+        self.sample_row_keys_result = []
+        self.regions_result = []
```
[the remark this answers is not in the document]
Comment (pchauhan-qlogic) [reply unit_tests/test_table.py]: done

File: unit_tests/test_table.py (hunk 4 above)
Comment (pchauhan-qlogic) [reply unit_tests/test_table.py]: added new test case https://github.com/googleapis/google-cloud-python-happybase/pull/70/commits/8220b6bf6454c43e5822743a89d10d0980505248#diff-ac84df270fae3be33d72fa312714fa13R164

**Comment by parthea** [comment]
I'm going to close this PR as stale as it's 2 years old but please feel free to re-open it.
